### ISSUE TITLE
modules/common.nix wants "inputs" as input, which seems at least odd?

### ISSUE THREAD

**Issue by lierdakil** [serokell/serokell.nix#101]
https://github.com/serokell/serokell.nix/blob/f179f81b4421eb3158af2cfe062adb9fce427b17/modules/common.nix#L5

So... AFAIU, modules [don't generally have input "inputs"](https://nixos.wiki/wiki/NixOS_modules#Function), so this creates issues downstream. Furthermore, while it's possible to work around by supplying "inputs", what "inputs" does it actually expect? I assume it expects this flake's inputs, not some random inputs from who knows where.

All that considered, shouldn't this rather be

```nix
inputs:  { lib, pkgs, config, options, ... }: 
```

and called from

https://github.com/serokell/serokell.nix/blob/f179f81b4421eb3158af2cfe062adb9fce427b17/flake.nix#L35

as

```nix
common = import ./modules/common.nix inputs;
```

or 

```nix
common = import ./modules/common.nix self.inputs;
```

?

**Comment by PhilTaken** [comment]
I agree the latter is a better and cleaner style. The former makes it easier to override certain inputs for the module / provide your own inputs but nowadays that can also be achieved with flake input overrides.

**Comment by rvem** [comment]
Resolved by #121
